llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -3837,11 +3837,11 @@ void MachineVerifier::verifyStackFrame() {
       BBState.ExitIsSetup = BBState.EntryIsSetup;
     }
 
-    if ((int)MBB->getCallFrameSize() != -BBState.EntryValue) {
+    if ((int)MBB->getCallFrameSizeOrZero() != -BBState.EntryValue) {
```

Comment:
Could you improve the verification here by making EntryValue/ExitValue be std::optionals too, and using the to track the presence/absence of a call frame?

Reply:
There was no good reason. It was just an oversight. If no tests failed, then I would not have noticed that there was a problem.

Reply:
Alright, I'll look for other such cases then and see if I can fix them. Thanks for the insight!

Reply:
I've updated the PR to a version with the adjusted MachineVerifier and fixes for the lit tests that failed with it. However, it's difficult to say whether this captures all cases where a CALLSEQ might be split across MachineBBs. For instance, I found 126 references (in 62 files) to MachineBasicBlock::splice; I think we'd have to handle them (and other cases, e.g., where new MachineBBs are created) or check that they can't affect CALLSEQs to be really sure we caught everything.
An alternative would be to drop the stored call frame size from the MachineBBs and compute up-to-date call frame sizes only when they are needed, i.e., in the PrologueEpilogueInserter and where I use it in #106965.
What's your opinion on that?

Reply:
> An alternative would be to drop the stored call frame size from the MachineBBs and compute up-to-date call frame sizes only when they are needed, i.e., in the PrologueEpilogueInserter

The whole point of storing the frame size in `MachineBB` is so that `PEI` does not have to do a complete forward walk over the function to compute frame sizes, in addition to the backward walk that it does to eliminate frame indices. My assumption was that the forward walk would be expensive, but if you can do it without angering https://llvm-compile-time-tracker.com/ then go for it!

Reply:
@jayfoad I just updated this PR to (re)compute the call frame sizes on demand instead of storing them. The compile-time overhead seems to be relatively minor ([see the CT report of the entire PR stack here](https://llvm-compile-time-tracker.com/compare.php?from=34b10e165d809bb133d37dfe934859800f21a100&to=1bd72c4ba6c98ced84b3d7d372dc8a3bc7aee7e9&stat=instructions%3Au)).

Reply:
Ping, @jayfoad